src/SharedStreamManager.mo

```diff
@@ -36,11 +44,19 @@ module {
     // info about each issued stream id is preserved here forever. Index is a stream ID
     let streams_ : Vec.Vector<StreamInfo<T>> = Vec.new();
     // a mapping of canister principal to stream id
-    var sourceCanistersStreamMap : AssocList.AssocList<Principal, ?Nat> = null;
+    var sourceCanistersStreamMap : AssocList.AssocList<Principal, ?Nat> = listFromIter<(Principal, ?Nat)>(
+      Iter.map<Principal, (Principal, ?Nat)>(
+        Iter.fromArray(initialSourceCanisters),
+        func(p) = (p, null),
+      )
+    );
```

Comment:
I think we should not create an `AssocList` directly, i.e. via the `List` methods. If we use AssocList.replace instead then we avoid accidental repetitions.

`initialSourceCanisters` could have repetitions. We have two options: 1) handle them gracefully, i.e. remove them, which happens automatically if we iterate over `initialSourceCanisters` and call `AssocList.replace` for each of them. 2) detect them and trap. Both options are fine.

Either way, I think the best is to iterate over the input and call AssocList.replace for each item.

Reply:
Fixed